### DISCUSSION TITLE
DEV UI in prod?

### DISCUSSION THREAD

**Discussion by PyAntony** [quarkusio/quarkus#24266] [Q&A]
Hi,

can I use the DEV UI (`/q/dev`) in prod? is there any way to activate it?

**Comment by maxandersen** [comment]
No. Devmode uses dev time features and liberties which if used during production you'll waste memory. 

Any specific features you are after ?

**Comment by PyAntony** [comment]
No specific features, we just like the DEV UI in general. We are used to it and we just realized we don't have access to it in prod. We would like it in prod, even if we have to pay some additional resources. So there is just no way at all to activate it in prod? 

(We have a few internal services that will be used by very few users, mostly other developers. Having the DEV UI available would be a GREAT addition for these services. It would be a nice feature to have the option to activate in prod.)

**Comment by GavinRay97** [comment]
I actually just went to the Zulip to ask this same thing:

https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/DevMode.20in.20Docker.20image

> I have a scenario in which other people in my team want to run our Quarkus app without setting up a development toolchain
> This is for local testing purposes
>
> It would be really useful for the Dev Mode UI to be available (particularly the live logs)

I don't mind if the image isn't marked "production", but some way to build a Docker image that Quarkus thinks is still Devmode would be really useful for observability + debuggability purposes.

**Reply by maxandersen** [reply]
well, have a dockerfile that copies the source into it and run `./mvnw quarkus:dev` would "just work" afaics?

**Reply by GavinRay97** [reply]
Huh, I don't see why not?

It would even have hot reload without the need for `gradle quarkusRemoteDev` (which is pretty broken with Kotlin from what I've experienced) if you mounted a volume with the source into it.

Something like this I suppose?

```dockerfile
FROM ghcr.io/graalvm/jdk:java17-22.0.0

WORKDIR /usr/src/app
COPY . .

CMD ["./gradlew", "quarkusDev"]
```

**Reply by PyAntony** [reply]
I always knew we could do something as simple as this. I just wanted to know if there exists a way to launch everything as "production" and activate the quarkus dev ui additionally (maybe there was a config like "quarkus.dev.ui.disable=false"). But yeah, this should work just fine.
